python/tskit/drawing.py

```diff
@@ -221,7 +235,355 @@ def remap_style(original_map, new_key, none_value):
+
+    @property
+    def max_y(self):
+        return self.total_size[1]
+
+    @property
+    def top(self):  # Alias for consistency with top & bottom
+        return self.pad_top
+
+    @property
+    def left(self):  # Alias for consistency with top & bottom
+        return self.pad_left
+
+    @property
+    def bottom(self):
+        return self.max_y - self.pad_bottom
+
+    @property
+    def right(self):
+        return self.max_x - self.pad_right
+
+    @property
+    def width(self):
+        return self.right - self.left
+
+    @property
+    def height(self):
+        return self.bottom - self.top
+
+    def __post_init__(self):
+        if self.width < 1 or self.height < 1:
+            raise ValueError("Image size too small to fit")
+
+    def draw(self, dwg, add_to, colour="grey"):
+        # used for debugging
+        add_to.add(
+            dwg.rect(
+                (0, 0),
+                (self.max_x, self.max_y),
+                fill="white",
+                fill_opacity=0,
+                stroke=colour,
+                stroke_dasharray="15,15",
+                class_="outer_plotbox",
+            )
+        )
+        add_to.add(
+            dwg.rect(
+                (self.left, self.top),
+                (self.width, self.height),
+                fill="white",
+                fill_opacity=0,
+                stroke=colour,
+                stroke_dasharray="5,5",
+                class_="inner_plotbox",
+            )
+        )
+
+
+class SvgPlot:
+    """ The base class for plotting either a tree or a tree sequence as an SVG file"""
+
+    standard_style = (
+        ".tree-sequence .background path {fill: #808080; fill-opacity:0}"
+        ".tree-sequence .background path:nth-child(odd) {fill-opacity:.1}"
+        ".axes {font-size: 14px}"
+        ".x-axis .tick .lab {font-weight: bold}"
+        ".axes, .tree {font-size: 14px; text-anchor:middle}"
+        ".y-axis line.grid {stroke: #FAFAFA}"
+        ".y-axis > .lab text {transform: translateX(0.8em) rotate(-90deg)}"
+        ".x-axis .tick g {transform: translateY(0.9em)}"
+        ".x-axis > .lab text {transform: translateY(-0.8em)}"
+        ".axes line, .edge {stroke:black; fill:none}"
+        ".node > .sym {fill: black; stroke: none}"
+        ".site > .sym {stroke: black}"
+        ".mut text {fill: red; font-style: italic}"
+        ".mut line {fill: none; stroke: none}"  # Default hide mut line to expose edges
+        ".mut .sym {fill: none; stroke: red}"
+        ".node .mut .sym {stroke-width: 1.5px}"
+        ".tree text, .tree-sequence text {dominant-baseline: central}"
+        ".plotbox .lab.lft {text-anchor: end}"
+        ".plotbox .lab.rgt {text-anchor: start}"
+    )
+
+    # TODO: we may want to make some of the constants below into parameters
+    text_height = 14  # May want to calculate this based on a font size
+    line_height = text_height * 1.2  # allowing padding above and below a line
+    root_branch_fraction = (
```

Comment:
This looks odd - put the comment on the line above?

Reply:
Oh, this is Black being silly. I'll rephrase.